Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+                return what
+            else:
+                return np.atleast_2d(what).T
+
+        # if anything is sparse, use the sparse version
+        from .impl import SparseTable
+        if sp.issparse(X) or (Y is not None and sp.issparse(Y)) \
+                          or (metas is not None and sp.issparse(metas)) \
+                          or (weights is not None and sp.issparse(weights))\
+                          or issubclass(cls, SparseTable):
+            # explicitly construct a sparse table as this can be called from Table(...),
+            # but allow subclasses to call correctly
+            if issubclass(cls, SparseTable):
+                return cls._from_sparse_numpy(domain, X, Y, metas, weights)
+            else:
+                return SparseTable._from_sparse_numpy(domain, X, Y, metas, weights)
+
+        if domain is None:
+            result = cls._from_data_inferred(X, Y, metas)
+            if weights is not None:
+                result.set_weights(weights)
+            return result
+
+        # ensure correct shapes (but not sizes) so we can iterate
+        X = correct_shape(X)
+        Y = correct_shape(Y)
+        metas = correct_shape(metas)
+
+        result = cls()
+        for role_array, variables in zip((X, Y, metas),
+                                         (domain.attributes, domain.class_vars, domain.metas)):
+            if role_array is None:
+                if variables:
+                    raise ValueError("Variable and column count mismatch. ")
+                continue
+            if role_array.shape[1] != len(variables):
+                raise ValueError("Variable and column count mismatch. ")
+            for column, variable in zip(role_array.T, variables):
+                result[variable.name] = column
+        result.domain = domain
+        if weights is not None:
+            result.set_weights(weights)
+
+        # transform any values we believe are null into actual null values
+        result.replace(to_replace=list(Variable.MISSING_VALUES), value=np.nan, inplace=True)
+        result._transform_discrete_values_to_descriptors()
+        result._transform_discrete_into_categorical()
+        result._transform_timevariable_into_datetime()
+
+        return result
+
+    @classmethod
+    def from_list(cls, domain, rows, weights=None):
+        """Construct a table from a list of rows.
+
+        Parameters
+        ----------
+        domain : Domain
+            The domain to use for the data.
+        rows : list[list[obj]]
+            A list of rows. Must be rectangular with the number of columns
+            matching the columns in the domain.
+        weights : anything TableBase.set_weights accepts, optional, default None
+            The weights to use for the resulting Table.
+
+        Returns
+        -------
+        TableBase
+            A Table constructed from the given data.
+        """
+        if weights is not None and len(rows) != len(weights):
```

Comment:
No, it isn't. This allows creating a mixed-type table from a list. Mixing strings and numbers, then converting to a `ndarray`, results in an `object` dtype which pandas uses for the whole thing. Passing a list doesn't have that problem.